resources/packs/gcp/gcp.lr

```diff
@@ -1040,4 +1042,112 @@ private gcp.project.cluster.nodepool.config.confidentialNodes @defaults("enabled
   id string
   // Whether to use confidential nodes
   enabled bool
-}
\ No newline at end of file
+}
+
+// GCP KMS resources
+private gcp.project.kms {
+  // Project ID
+  projectId string
+  // The available locations for the service
+  locations() []string
+  // The list of keyrings in the current project
+  keyrings() []gcp.project.kms.keyring
+}
+
+// GCP KMS keyring
+private gcp.project.kms.keyring @defaults("name") {
+  // Internal ID
+  id string
+  // Project ID
+  projectId string
+  // The name of the keyring
+  name string
+  // Creation timestamp
+  created time
+  // The location of the keyring
+  location string
+  // The list of cryptokeys in the current keyring
+  cryptokeys() []gcp.project.kms.keyring.cryptokey
+}
+
+// GCP KMS crypto key
+private gcp.project.kms.keyring.cryptokey @defaults("name purpose"){
+  // Internal ID
+  id string
+  // The name of the crypto key
+  name string
+  // The primary version that will be used by encrypt when this crypto key is given
+  primary gcp.project.kms.keyring.cryptokey.version
+  // The purpose of the crypto key
+  purpose string
+  // The list of cryptokey versions
+  versions() []gcp.project.kms.keyring.cryptokey.version
+}
+
+// GCP KMS crypto key version
+private gcp.project.kms.keyring.cryptokey.version @defaults("name state") {
+  // Internal ID
+  id string
+  // The name of the crypto key version
+  name string
+  // The current state of the crypto key version
+  state string
+  // The protection level describing how crypto operations are performed with this crypto key version
+  protectionLevel string
+  // The algorithm that this crypto key version supports
+  algorithm string
+  // Statement generated and signed by HSM at key creation time
+  attestation gcp.project.kms.keyring.cryptokey.version.attestation
+  // Creation timestamp
+  created time
+  // Generation timestamp
```

Comment:
```suggestion
  // Time generated
```
Hey @imilchev what's the difference between created and generated? And should we tell our users?

Reply:
This is what Google's own docs say:
```go
        // Output only. The time at which this
	// [CryptoKeyVersion][google.cloud.kms.v1.CryptoKeyVersion] was created.
	CreateTime *timestamppb.Timestamp `protobuf:"bytes,4,opt,name=create_time,json=createTime,proto3" json:"create_time,omitempty"`
	// Output only. The time this
	// [CryptoKeyVersion][google.cloud.kms.v1.CryptoKeyVersion]'s key material was
	// generated.
	GenerateTime *timestamppb.Timestamp `protobuf:"bytes,11,opt,name=generate_time,json=generateTime,proto3" json:"generate_time,omitempty"`
```

not entirely sure what it means as I don't know much about this service